### PR TITLE
Add numerous POM improvements, adjust summarize.sh, add award to README.md 

### PR DESCRIPTION
**Pull request by korotkevics**
* Add explanations on Antrun, URL, Assembly, Plugin Management
* Add Checkstyle, Enforcer, JaCoCo to build
* Add *.orig to .gitignore
* Fix a Checkstyle warning
* Adjust server module description
* Add a script which generates a local site
* Move common plugins' properties to pluginManagement (Lombok, TestNG, Mockito)
* Update README.md with a new award (chat)
* Replace default Maven skin with Fluido
* Add public method count to summarize.sh

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -227,11 +243,37 @@
...
                 </plugin>
+                <!--
+                We were supposed to tell, how Maven Antrun Plug-in
+                was used.
+                Maven Antrun Plug-in can be used to run
+                Ant tasks. We do not know much about Ant, and
+                the most comprehensive/useful example we've found was
+                running a script with a build phase:
+                https://maven.apache.org/guides/mini/guide-using-ant.html
+                We haven't used it in our project anyways because
```
Comment (szczepanskikrs) [inline pom.xml]: Anyways -> anyway?

File: pom.xml [hunk 2]
```diff
@@ -251,28 +293,107 @@
...
+                        </execution>
+                        <execution>
+                            <id>check</id>
+                            <goals>
+                                <goal>check</goal>
+                            </goals>
+                            <configuration>
+                                <rules>
+                                    <!--SHARED module only
+                                    TODO: increase, so the cross-module average can be 60%.
```
Comment (szczepanskikrs) [inline pom.xml]: Please consider removing TODO: item.

File: pom.xml [hunk 3]
```diff
@@ -251,28 +293,107 @@
...
+                                        <limits>
+                                            <limit implementation="org.jacoco.report.check.Limit">
+                                                <counter>INSTRUCTION</counter>
+                                                <value>COVEREDRATIO</value>
+                                                <minimum>0.68</minimum>
+                                            </limit>
+                                        </limits>
+                                    </rule>
+                                    <!--SERVER module only
+                                    TODO: increase, so the cross-module average can be 60%.
```
Comment (szczepanskikrs) [inline pom.xml]: As above

File: pom.xml [hunk 4]
```diff
@@ -449,6 +570,17 @@
     <distributionManagement>
         <site>
             <id>${project.artifactId}</id>
+            <!--
+            We deploy site with a script (updateRemoteSite.sh),
+            therefore scp is left empty.
```
Comment (szczepanskikrs) [inline pom.xml]: what is scp?

File: pom.xml (hunk 1 above)
Comment (korotkevics) [reply pom.xml]: You're right, anyways is slang, I didn't know that:) http://www.gingersoftware.com/content/blog/anyway-vs-anyways/

File: pom.xml (hunk 4 above)
Comment (korotkevics) [reply pom.xml]: the thing present in the url:) Frankly, I haven't found what the abbreviation meant, but normally there is an URL in between, ex. <url>scp://www.google.com/</url>

File: pom.xml (hunk 1 above)
Comment (korotkevics) [reply pom.xml]: I'll fix it right away and introduce it in future pull requests

**Pull request merged**